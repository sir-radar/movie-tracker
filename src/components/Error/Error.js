import React from 'react'

const Error = () => {
  return (
  <section data-testid="error" className='loading mt-5 mx-auto text-center'>
    <i className="fa fa-times fa-5x text-danger"></i>
    <h1 data-testid="error-header">Opps</h1>
    <h3 data-testid="error-body">Something went wrong. Please check your internet and try again.</h3>
  </section>
  )
}

export default Error