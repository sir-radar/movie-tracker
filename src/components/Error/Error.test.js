import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Error from './Error';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

afterEach(cleanup)

const renderComponent = () => {
  const div = document.createElement("div");
  return render(<Error/>, div)
}

it("renders without crashing", () => {
  renderComponent()
})

it("renders page texts correctly", () => {
  const { getByTestId } = renderComponent()
  expect(getByTestId('error-header')).toHaveTextContent("Opps")
  expect(getByTestId('error-body')).toHaveTextContent("Something went wrong. Please check your internet and try again")
})

it("matches snapshot", () => {
  const tree = renderer.create(<Error/>).toJSON();
  expect(tree).toMatchSnapshot();
})