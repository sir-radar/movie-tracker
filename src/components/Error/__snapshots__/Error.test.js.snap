// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="loading mt-5 mx-auto text-center"
  data-testid="error"
>
  <i
    className="fa fa-times fa-5x text-danger"
  />
  <h1
    data-testid="error-header"
  >
    Opps
  </h1>
  <h3
    data-testid="error-body"
  >
    Something went wrong. Please check your internet and try again.
  </h3>
</section>
`;
